controller/volume_controller.go

```diff
@@ -1141,10 +1142,120 @@ func (vc *VolumeController) ReconcileVolumeState(v *longhorn.Volume, es map[stri
 		if oldState != v.Status.State {
 			vc.eventRecorder.Eventf(v, v1.EventTypeNormal, EventReasonAttached, "volume %v has been attached to %v", v.Name, v.Status.CurrentNodeID)
 		}
+
+		keepLocalReplicaPolicy, err := vc.ds.GetSettingAsBool(types.SettingNameKeepLocalReplicaOnSameNodeAsEngine)
+		if err != nil {
+			return err
+		}
+		if keepLocalReplicaPolicy {
+			if err := vc.handleKeepLocalReplicaOnSameNodeAsEngine(v, e, rs); err != nil {
+				return err
+			}
+		}
 	}
 	return nil
 }
 
+func (vc *VolumeController) handleKeepLocalReplicaOnSameNodeAsEngine(v *longhorn.Volume, e *longhorn.Engine, rs map[string]*longhorn.Replica) error {
```

Comment:
This function naming is a little bit weird. I think a simpler naming would be better. e.g., `ReconcileLocalReplica`

Reply:
Thanks. Fixed